pkg/controllers/jobset_controller_test.go

```diff
@@ -700,6 +704,223 @@ func TestUpdateConditions(t *testing.T) {
+						jobName:           "test-jobset-replicated-job-2-test-job-0",
+						ns:                ns,
+						replicas:          3,
+						jobIdx:            0,
+						parallelism:       pointer.Int32(5),
+						ready:             pointer.Int32(2),
+						succeeded:         3}).Obj(),
+					makeJob(&makeJobArgs{
+						jobSetName:        jobSetName,
+						replicatedJobName: "replicated-job-2",
+						jobName:           "test-jobset-replicated-job-2-test-job-1",
+						ns:                ns,
+						replicas:          3,
+						jobIdx:            0,
+						parallelism:       pointer.Int32(3),
+						completions:       pointer.Int32(2),
+						ready:             pointer.Int32(1),
+						succeeded:         1}).Obj(),
+					makeJob(&makeJobArgs{
+						jobSetName:        jobSetName,
+						replicatedJobName: "replicated-job-2",
+						jobName:           "test-jobset-replicated-job-2-test-job-2",
+						ns:                ns,
+						replicas:          3,
+						jobIdx:            0,
+						parallelism:       pointer.Int32(2),
+						completions:       pointer.Int32(3),
+						ready:             pointer.Int32(2),
+						succeeded:         1}).Obj(),
+				},
+			},
+			expected: []jobset.ReplicatedJobStatus{
+				{
+					Name:          "replicated-job-1",
+					ReadyJobs:     1,
+					SucceededJobs: 0,
+				},
+				{
+					Name:          "replicated-job-2",
+					ReadyJobs:     3,
+					SucceededJobs: 0,
+				},
+			},
+		},
+		{
+			name: "no jobs created",
+			js: testutils.MakeJobSet(jobSetName, ns).
+				ReplicatedJob(testutils.MakeReplicatedJob("replicated-job-1").
+					Job(testutils.MakeJobTemplate("test-job", ns).Obj()).
+					Replicas(1).
+					Obj()).
+				ReplicatedJob(testutils.MakeReplicatedJob("replicated-job-2").
+					Job(testutils.MakeJobTemplate("test-job", ns).Obj()).
+					Replicas(3).
+					Obj()).Obj(),
+			expected: []jobset.ReplicatedJobStatus{
+				{
+					Name:          "replicated-job-1",
+					ReadyJobs:     0,
+					SucceededJobs: 0,
+				},
+				{
+					Name:          "replicated-job-2",
+					ReadyJobs:     0,
+					SucceededJobs: 0,
+				},
+			},
+		},
+		{
+			name: "partial jobs created",
+			js: testutils.MakeJobSet(jobSetName, ns).
+				ReplicatedJob(testutils.MakeReplicatedJob("replicated-job-1").
+					Job(testutils.MakeJobTemplate("test-job", ns).Obj()).
+					Replicas(1).
+					Obj()).
+				ReplicatedJob(testutils.MakeReplicatedJob("replicated-job-2").
+					Job(testutils.MakeJobTemplate("test-job", ns).Obj()).
+					Replicas(3).
+					Obj()).Obj(),
+			jobs: childJobs{
+				active: []*batchv1.Job{
+					makeJob(&makeJobArgs{
+						jobSetName:        jobSetName,
+						replicatedJobName: "replicated-job-2",
+						jobName:           "test-jobset-replicated-job-2-test-job-0",
+						ns:                ns,
+						replicas:          3,
+						jobIdx:            0,
+						parallelism:       pointer.Int32(5),
+						ready:             pointer.Int32(2),
+						succeeded:         3}).Obj(),
+					makeJob(&makeJobArgs{
```

Comment:
just to reduce the length of the file, I think one job here is enough (instead of 3)